src/main/kotlin/com/angorasix/projects/management/core/infrastructure/persistence/repository/ProjectManagementFilterRepositoryImpl.kt

```diff
@@ -20,10 +22,31 @@ class ProjectManagementFilterRepositoryImpl(val mongoOps: ReactiveMongoOperation
     override fun findUsingFilter(filter: ListProjectsManagementFilter): Flow<ProjectManagement> {
         return mongoOps.find(filter.toQuery(), ProjectManagement::class.java).asFlow()
     }
+
+    override suspend fun findByIdForContributor(filter: ListProjectsManagementFilter,
+                                                simpleContributor: SimpleContributor?): ProjectManagement? {
+        return mongoOps.find(filter.toQuery(simpleContributor), ProjectManagement::class.java)
+                .awaitFirstOrNull()
+    }
 }
 
-private fun ListProjectsManagementFilter.toQuery(): Query {
+private fun ListProjectsManagementFilter.toQuery(simpleContributor: SimpleContributor? = null): Query {
     val query = Query()
+
+    val requestingOthers = adminId == null || !adminId.contains(simpleContributor?.contributorId)
+    val requestingOwn =
+            simpleContributor != null && (adminId.isNullOrEmpty() || adminId.contains(simpleContributor.contributorId))
+
+    ids?.let { query.addCriteria(where("_id").`in`(it)) }
     projectIds?.let { query.addCriteria(where("projectId").`in`(it)) }
+
+    if (requestingOthers) {
+        return query
+    }
+
+    if (requestingOwn) {
+        query.addCriteria(where("admins").elemMatch(where("contributorId").`is`(simpleContributor?.contributorId)))
+    }
```

Comment:
If this comment makes sense:
https://github.com/AngoraSix/projects.presentations/pull/22#discussion_r1444159204
Then we probably want to simplify the logic similarly here :)